### PR TITLE
feat(ui): Group nodes by column values

### PR DESCRIPTION
**Pull request by ahochsteger**
This PR adds the possibility to group the nodes by column values.

Grouping can be enabled in the column filter options:
image

Grouped values are shown like this (using v-data-table grouping feature):
image

The groups can be collapsed (click on "-") and removed (click on "X").

### CODE REVIEW

**Review by robertsLando — CHANGES_REQUESTED** [review]
Nice feature! Thanks @ahochsteger !

File: src/components/nodes-table/nodes-table.js [hunk 1]
```diff
@@ -66,12 +81,21 @@ export default {
     }
   },
   created () {
+    this.showHidden = this.settings.load('nodes_showHidden', false)
     this.filters = this.loadSetting('nodes_filters', this.initFilters())
     this.sorting = this.loadSetting('nodes_sorting', this.initSorting())
-    this.nodeTableItems = this.loadSetting('nodes_itemsPerPage', 10)
+    this.groupBy = this.loadSetting('nodes_groupBy', [])
+    console.log('created(): groupBy=', this.groupBy)
```
[the remark this answers is not in the document]
Comment (ahochsteger) [reply src/components/nodes-table/nodes-table.js]: Ooops, that slipped through after debugging - it's gone now.
Thanks for reporting!

**Comment by robertsLando** [comment]
@ahochsteger Is hot reloading working on your side? (it's not about this PR)

**Review by robertsLando — CHANGES_REQUESTED** [review]
I'm getting this error on console:

```
Vue warn]: Property or method "nodeTableItems" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.

found in

---> <NodesTable> at src/components/nodes-table/index.vue
       <VCard>
         <ControlPanel> at src/components/ControlPanel.vue
           <VMain>
             <VApp>
               <App> at src/App.vue
                 <Root>
```

**Comment by ahochsteger** [comment]
> I'm getting this error on console:

It's fixed now - was a missing rename.
Strange that it didn't report that on my side.

May be related to your question about hot-reloading.
It seems that it's not working anymore - maybe since I updated my branch to the latest master.

This is what I just got which required me to answer with **yes**:
```bash
$ npm run dev

> zwavejs2mqtt@1.0.0-alpha.2 dev /home/a13870/oeamtc/ws/github/zwave-js/zwavejs2mqtt-clean
> webpack serve --inline --progress --host 0.0.0.0 --config build/webpack.dev.conf.js

The command moved into a separate package: @webpack-cli/serve
Would you like to install serve? (That will run npm install -D @webpack-cli/serve) (yes/NO) : yes
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
...
```

**Comment by robertsLando** [comment]
> This is what I just got which required me to answer with yes:

Yes but if you change something is the change hot reloaded on the page or does you need to refre3sh the page to see the new edist? I think it's something broken after the move to webpack 5 and I'm not able to find the reason. When I chage something on the code I see that the teminal show a reload but the page is not reloaded

**Comment by robertsLando** [comment]
I think you may need to run an `npm install` command as some of your deps could be outdated

**Review by robertsLando — APPROVED** [review]
LGTM

**Comment by ahochsteger** [comment]
> I think you may need to run an `npm install` command as some of your deps could be outdated

Even after `npm install` hot-reloading of changes in .vue files does not work anymore.
It worked on this branch before I merged the latest changes from master.

**Comment by robertsLando** [comment]
> It worked on this branch before I merged the latest changes from master.

It's something broken after #155 I think

**Comment by ahochsteger** [comment]
> It's something broken after #155 I think

Ok, I'll leave a comment there, as removing 'node_modules' reveals an error for me ...

**Comment by robertsLando** [comment]
@ahochsteger If you can, give it a look. I tried to fix it some days ago without success. I had to move from webbac-dev-server to webpack serve command and everything seems working (no error shown) but no hot reloading

**Comment by ahochsteger** [comment]
> @ahochsteger If you can, give it a look. I tried to fix it some days ago without success. I had to move from webpack-dev-server to webpack serve command and everything seems working (no error shown) but no hot reloading

Nevermind, the error I got was due to open file watches in VSCode that prevented old files from node_modules to be removed.
Closing VSCode, removing node_modules, then do 'npm install' at least does not run into an error anymore on 'npm run dev'.

I'm not sure, if I can help with the hot-reloading though, this is something I've not been that deep into yet.

**Comment by robertsLando** [comment]
> I'm not sure, if I can help with the hot-reloading though, this is something I've not been that deep into yet.

No worries, I'm not a webpack expert too, I will look at it next days :) thanks

**Pull request merged**